### PR TITLE
Move all portable Profile7 projects to netstandard1.3

### PR DESCRIPTION
**Pull request by agocke**
This change drops support for .NET 4.5 and retargets to desktop
4.6 + CoreCLR. It also fixes up various analyzer warnings that
were created due to the move.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/project.json [hunk 1]
```diff
@@ -1,6 +1,9 @@
 {
-  "dependencies": { },
+  "dependencies": {
+    "System.Linq.Expressions": "4.1.0",
+    "System.Threading.Tasks.Parallel": "4.0.1"
+  },
   "frameworks": {
-    ".NETPortable,Version=v4.5,Profile=Profile7": {}
+    "netstandard1.3": {}
```
Comment (tannergooding) [inline src/Compilers/CSharp/Portable/project.json]: Where is the reference to `NETStandard.Library` and `Microsoft.NetCore.Portable.Compatibility` flowing from?

File: src/Compilers/Core/Portable/CodeAnalysis.csproj [hunk 2]
```diff
@@ -13,14 +13,14 @@
     <AssemblyName>Microsoft.CodeAnalysis</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\..\</SolutionDir>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
     <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
```
Comment (tannergooding) [inline src/Compilers/Core/Portable/CodeAnalysis.csproj]: VS keeps `<TargetFrameworkProfile></TargetFrameworkProfile>` (but has it empty), we should probably do the same.

File: build/Targets/ProducesNoOutput.Settings.targets [hunk 3]
```diff
@@ -19,8 +19,7 @@
 
   <PropertyGroup Condition="'$(IsPortable)' == 'true'">
     <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>    
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
```
Comment (tannergooding) [inline build/Targets/ProducesNoOutput.Settings.targets]: Should have empty `TargetFrameworkProfile`

File: src/Compilers/CSharp/Portable/CSharpCodeAnalysis.csproj [hunk 4]
```diff
@@ -13,9 +13,8 @@
     <AssemblyName>Microsoft.CodeAnalysis.CSharp</AssemblyName>
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\..\</SolutionDir>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
     <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
```
Comment (tannergooding) [inline src/Compilers/CSharp/Portable/CSharpCodeAnalysis.csproj]: Should have empty `TargetFrameworkProfile`

File: src/Compilers/Core/Portable/FileSystem/PathUtilities.cs [hunk 5]
```diff
@@ -21,7 +22,7 @@ internal static bool IsUnixLikePlatform
         {
             get
             {
-                return PortableShim.Path.DirectorySeparatorChar == '/';
+                return Path.DirectorySeparatorChar == '/';
```
Comment (tannergooding) [inline src/Compilers/Core/Portable/FileSystem/PathUtilities.cs]: I thought there was a new RuntimeInformation API that determined this?

File: src/Compilers/Core/Portable/project.json [hunk 6]
```diff
@@ -1,8 +1,26 @@
 ﻿{
   "supports": {},
   "dependencies": {
+    "System.Collections": "4.0.11",
```
Comment (tannergooding) [inline src/Compilers/Core/Portable/project.json]: Most of these should come from `NETStandard.Library` and `Microsoft.NETCore.Portable.Compatibility`

File: src/Compilers/Test/Resources/Core/CompilerTestResources.csproj [hunk 7]
```diff
@@ -13,8 +13,7 @@
     <AssemblyName>Roslyn.Compilers.Test.Resources</AssemblyName>
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\..\..\</SolutionDir>
     <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
```
Comment (tannergooding) [inline src/Compilers/Test/Resources/Core/CompilerTestResources.csproj]: Should have empty `TargetFrameworkProfile`

File: src/Compilers/Test/Resources/Core/project.json [hunk 8]
```diff
@@ -1,7 +1,9 @@
 {
   "supports": { },
-  "dependencies": { },
+  "dependencies": {
+    "System.Xml.ReaderWriter": "4.0.11"
```
Comment (tannergooding) [inline src/Compilers/Test/Resources/Core/project.json]: Need reference to `NETStandard.Library` and `Microsoft.NETCore.Portable.Compatibility`

File: src/Compilers/VisualBasic/Portable/BasicCodeAnalysis.vbproj [hunk 9]
```diff
@@ -11,9 +11,8 @@
     <OutputType>Library</OutputType>
     <AssemblyName>Microsoft.CodeAnalysis.VisualBasic</AssemblyName>
     <SolutionDir Condition="'$(SolutionDir)' == '' OR '$(SolutionDir)' == '*Undefined*'">..\..\..\..\</SolutionDir>
-    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
-    <TargetFrameworkProfile>Profile7</TargetFrameworkProfile>
     <TargetFrameworkIdentifier>.NETPortable</TargetFrameworkIdentifier>
+    <TargetFrameworkVersion>v5.0</TargetFrameworkVersion>
```
Comment (tannergooding) [inline src/Compilers/VisualBasic/Portable/BasicCodeAnalysis.vbproj]: Should have empty `TargetFrameworkProfile`

File: src/Compilers/VisualBasic/Portable/project.json [hunk 10]
```diff
@@ -1,6 +1,8 @@
 {
-  "dependencies": { },
+  "dependencies": {
+    "System.Threading.Tasks.Parallel": "4.0.1"
```
Comment (tannergooding) [inline src/Compilers/VisualBasic/Portable/project.json]: Need a reference to `NETStandard.Library` and `Microsoft.NETCore.Portable.Compatibility`

File: src/Dependencies/Composition/project.json [hunk 11]
```diff
@@ -1,9 +1,12 @@
...
   "supports": {},
   "dependencies": {
-    "Microsoft.Composition": "1.0.27"
+    "Microsoft.Composition": "1.0.27",
+    "System.Runtime": "4.1.0"
   },
   "frameworks": {
-    ".NETPortable,Version=v4.5,Profile=Profile7": {}
+    "netstandard1.3": {
+      "imports": "portable-net45+win8"
```
Comment (tannergooding) [inline src/Dependencies/Composition/project.json]: portable-net46 is a valid TFM and covers this.

File: src/Dependencies/Composition/project.json [hunk 12]
```diff
@@ -1,9 +1,12 @@
 {
   "supports": {},
   "dependencies": {
-    "Microsoft.Composition": "1.0.27"
+    "Microsoft.Composition": "1.0.27",
+    "System.Runtime": "4.1.0"
```
Comment (tannergooding) [inline src/Dependencies/Composition/project.json]: This should flow from a reference to `NETStandard.Library` and `Microsoft.NETCore.Portable.Compatibility`

**Comment by tannergooding** [comment]
The changes look good to me. However, we should be using a reference to `NETStandardLibrary` (v1.6.0) and `Microsoft.NETCore.Portable.Compatibility` (v1.0.1). The primary motivation being that there are various IDE things that break if we don't.

Additionally, VS by default keeps the `TargetFrameworkProfile` definition, leaving it empty. We should probably do the same.

Also, we should probably be using the `dotnet5.4` TFM instead of `dotnet` and the `portable-net46` moniker instead of `portable-net45+win8`.

**Comment by agocke** [comment]
@tannergooding 

>However, we should be using a reference to NETStandardLibrary

Roslyn should *never* have a reference to `NETStandard.Library` as it is part of `NETStandard.Library.

> Microsoft.NETCore.Portable.Compatibility

I don't see the purpose. M.N.P.C is a shim between mscorlib -> System.Runtime. I believe we're fully System.Runtime now.

> Additionally, VS by default keeps the TargetFrameworkProfile definition, leaving it empty. We should probably do the same.

I see no good reason for this.

> Also, we should probably be using the dotnet5.4 TFM instead of dotnet and the portable-net46 moniker instead of portable-net45+win8.

I see no good reason for this either.

**Comment by agocke** [comment]
FYI @CyrusNajmabadi 

Jason told me you were looking to make some of these changes. I think this PR subsumes all the changes you wanted to make.

File: src/ExpressionEvaluator/CSharp/Source/ResultProvider/NetFX20/project.json [hunk 13]
```diff
@@ -1,8 +1,8 @@
 {
   "dependencies": {
     "Microsoft.NetFX20": "1.0.3",
-    "Microsoft.VisualStudio.Debugger.Engine": "14.3.25420",
-    "Microsoft.VisualStudio.Debugger.Metadata": "14.3.25420"
+    "Microsoft.VisualStudio.Debugger.Engine": "14.3.25421",
+    "Microsoft.VisualStudio.Debugger.Metadata": "14.3.25421"
```
Comment (tmat) [inline src/ExpressionEvaluator/CSharp/Source/ResultProvider/NetFX20/project.json]: Is this related change?

File: src/Features/CSharp/Portable/project.json [hunk 14]
```diff
@@ -2,6 +2,8 @@
   "dependencies": {
   },
   "frameworks": {
-    ".NETPortable,Version=v4.5,Profile=Profile7": {}
+    "netstandard1.3": {
+      "imports": [ "portable-net45+win8", "dotnet" ]
```
Comment (tmat) [inline src/Features/CSharp/Portable/project.json]: I think @jasonmalinowski removed multiple imports from everywhere to work around some issue.

File: src/Scripting/Core/Scripting.csproj [hunk 15]
```diff
@@ -47,6 +47,9 @@
     <Compile Include="..\..\Compilers\Core\Portable\FileSystem\RelativePathResolver.cs">
       <Link>Resolvers\RelativePathResolver.cs</Link>
     </Compile>
+    <Compile Include="..\..\Compilers\Shared\CoreClrShim.cs">
+      <Link>CoreClrShim.cs</Link>
```
Comment (tmat) [inline src/Scripting/Core/Scripting.csproj]: Is CoreClrShim still used elsewhere? If not, let's just move it here instead of linking it.

File: src/VisualStudio/Setup/VisualStudioSetup.csproj [hunk 16]
```diff
@@ -147,6 +147,32 @@
     <NuGetPackageToIncludeInVsix Include="Microsoft.Composition" />
     <NuGetPackageToIncludeInVsix Include="System.Reflection.Metadata" />
     <NuGetPackageToIncludeInVsix Include="System.Collections.Immutable" />
+    <NuGetPackageToIncludeInVsix Include="System.Collections" />
```
Comment (tmat) [inline src/VisualStudio/Setup/VisualStudioSetup.csproj]: Do we now need to insert more facades to VS? That might cause red flags in VS perf tests. @KevinH-MS

Comment (tmat) [inline src/VisualStudio/Setup/VisualStudioSetup.csproj]: Before we do that we should stop inserting them into ExternalAPI/Roslyn and instead insert each facade into a separate ExternalAPI/{facade-name} directory.

In addition we need to add appropriate binding redirect entries.

File: src/Workspaces/Core/Portable/Workspaces.csproj [hunk 17]
```diff
@@ -1022,4 +1012,9 @@
   <ImportGroup Label="Targets">
     <Import Project="..\..\..\..\build\Targets\VSL.Imports.targets" />
   </ImportGroup>
+  <ProjectExtensions>
+    <VisualStudio>
+      <UserProperties project_1json__JSONSchema="http://json.schemastore.org/nuget-project" />
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Workspaces.csproj]: Is this intentional?

**Comment by tannergooding** [comment]
@agocke, there are some IDE features that break if you don't reference `M.N.P.C`. This bug covers part of it: https://github.com/dotnet/roslyn/issues/11749, but basically, the error list and other things just break if we don't reference it on a Portable project targeting netstandard (@rchande may have more details, such as if this has or will be fixed in the IDE).

**Comment by crummel** [comment]
Build triggered. sha1 is original commit.

File: src/Features/CSharp/Portable/project.json (hunk 14 above)
Comment (jasonmalinowski) [inline src/Features/CSharp/Portable/project.json]: If you have multiple, it'll break fast up-to-date checks. @rchande is aware of the actual bug and can help you with it. The workaround is to only have one, if you can get away with it. If you really need multiple, well, @rchande might have another idea.

**Comment by agocke** [comment]
Test windows_release_unit64_prtest please
